src/modules/feature-modules/admin/pages/users/roles/role-new.config.ts

```diff
@@ -0,0 +1,219 @@
+import { AppInjector } from '@modules/core';
+
+import { UserRoleEnum } from '@app/base/enums';
+
+import { FormEngineModel, WizardEngineModel, WizardStepType, WizardSummaryType } from '@modules/shared/forms';
+import { AuthenticationStore } from '@modules/stores';
+
+// Labels.
+const stepsLabels = {
+  l1: { label: 'What is their role?' },
+  l2: { label: 'Which organisation is the user associated to?' },
+  l3: { label: 'Which unit is the user associated to?' }
+};
+
+
+// Types.
+type Organisation = { id: string; name: string; units: { id: string; name: string; }[] };
+type UserRole = { role: UserRoleEnum, orgId?: string, unitId?: string };
+type AvailableRoles = UserRoleEnum.QUALIFYING_ACCESSOR | UserRoleEnum.ACCESSOR | UserRoleEnum.ASSESSMENT;
+
+type InboundPayloadType = {
+  userRoles: UserRole[],
+  organisations?: Organisation[],
+
+  email: string,
+  name: string,
+  organisationId?: string,
+  unitId?: string
+};
+
+type LogicFieldsType = {
+  email: string;
+  name: string;
+
+  inboundData: { orgId?: string, unitId?: string },
+  organisations?: Organisation[];
+  availableRoles: UserRoleEnum[];
+  userRoles: UserRole[],
+
+}
+type QuestionFieldsType = {
+  role?: AvailableRoles;
+  organisationId?: string;
+  unitIds?: string[];
+}
+type StepPayloadType = LogicFieldsType & QuestionFieldsType;
+
+type CreateRolesType =
+  | { role: AvailableRoles }
+  | {
+    role: UserRoleEnum.ACCESSOR | UserRoleEnum.QUALIFYING_ACCESSOR;
+    organisationId: string;
+    unitIds: string[];
+  };
+export type OutboundPayloadType = CreateRolesType;
+
+// consts.
+const injector = AppInjector.getInjector();
```

Comment:
You don't need this I believe...

Reply:
I need it to inject `authenticationStore`

Reply:
Gotcha... ;)